### PR TITLE
check for existing template-config-file

### LINKED ISSUE #1106: beacon init does not error when providing a non-existent template config file
**Describe the bug**
When I run `yarn run cli beacon init     --eth1.enabled=false     --chain.genesisStateFile=genesis.ssz     --template-config-file altona.json     --root-dir .altona/`, I get no errors and the output from the cli just shows what it would normally show when init has been ran successfully.

**Expected behavior**
I would expect to see some sort of error telling the user to provide a .json file

**Steps to Reproduce**
`lerna clean`
`lerna bootstrap`
`yarn run cli beacon init     --eth1.enabled=false     --chain.genesisStateFile=genesis.ssz     --template-config-file altona.json     --root-dir .altona/`

### PR DESCRIPTION
**Pull request by 3xtr4t3rr3str14l**
-fixes #1106

### CODE REVIEW

**Comment by 3xtr4t3rr3str14l** [comment]
just realized a similar errors happens when providing a non-existent genesisStateFile, so i'm gonna push something for that in a second here too

**Review by wemeetagain — CHANGES_REQUESTED** [review]
I think we just need to remove the silly try catch from readBeaconConfig. We don't want to coerce here.

Edit: alternatively, add a file-exists check in src/cmds/beacon/cmds/init/index.ts

File: packages/lodestar-cli/src/cmds/beacon/cmds/init/index.ts [hunk 1]
```diff
@@ -13,8 +14,15 @@ export const description = "Initialize lodestar beacon node";
 export function builder(yargs: Argv<IBeaconArgs>): Argv<IBeaconArgs> {
   yargs.options(beaconInitOptions);
   const args = parseArgs(yargs) as IBeaconInitArgs;
-  if (args.templateConfigFile) {
-    yargs.config(readBeaconConfig(args.templateConfigFile));
+
+  if (args.chain && args.chain.genesisStateFile) {
+    fs.readFileSync(args.chain.genesisStateFile, "utf8");
+  }
+
```
Comment (wemeetagain) [inline packages/lodestar-cli/src/cmds/beacon/cmds/init/index.ts]: This check shouldn't be done here.
Check beacon/cmds/run/run.ts. Add a try/catch around the genesis state file `fs.promises.readFile` and catch, throw a YargsError with a descriptive error msg.

File: packages/lodestar-cli/src/cmds/beacon/cmds/init/index.ts [hunk 2]
```diff
@@ -13,8 +14,15 @@ export const description = "Initialize lodestar beacon node";
...
+  const {templateConfigFile} = args;
+  if (templateConfigFile) {
+    fs.readFileSync(templateConfigFile, "utf8");
+    yargs.config(readBeaconConfig(templateConfigFile));
```
Comment (wemeetagain) [inline packages/lodestar-cli/src/cmds/beacon/cmds/init/index.ts]: Can you investigate why there's a try/catch around `readBeaconConfig`? Likely should be removed as a better solution than this.

**Review by wemeetagain — CHANGES_REQUESTED** [review]

File: packages/lodestar-cli/src/cmds/beacon/cmds/init/index.ts (hunk 2 above)
Comment (3xtr4t3rr3str14l) [reply packages/lodestar-cli/src/cmds/beacon/cmds/init/index.ts]: yep.  just took out the try/catch

Comment (3xtr4t3rr3str14l) [reply packages/lodestar-cli/src/cmds/beacon/cmds/init/index.ts]: and took out my initial solution

File: packages/lodestar-cli/src/cmds/beacon/cmds/init/index.ts (hunk 1 above)
Comment (3xtr4t3rr3str14l) [reply packages/lodestar-cli/src/cmds/beacon/cmds/init/index.ts]: after this PR is done, i think i might try going after #1108 and maybe just include the fix for this in that PR instead

Comment (3xtr4t3rr3str14l) [reply packages/lodestar-cli/src/cmds/beacon/cmds/init/index.ts]: so for now, just gonna use this PR for template-config-file fix

**Review by mpetrunic — DISMISSED** [review]

File: packages/lodestar-cli/src/cmds/beacon/config.ts [hunk 3]
```diff
@@ -32,11 +32,7 @@ export async function writeBeaconConfig(filename: string, config: Partial<IBeaco
  * This needs to be a synchronous function because it will be run as part of the yargs 'build' step
  */
 export function readBeaconConfig(filename: string): Partial<IBeaconNodeOptions> {
-  try {
-    return readFileSync(filename) as Partial<IBeaconNodeOptions>;
-  } catch(e) {
-    return {};
-  }
+  return readFileSync(filename) as Partial<IBeaconNodeOptions>;
```
Comment (wemeetagain) [inline packages/lodestar-cli/src/cmds/beacon/config.ts:35]: I think this diff is the only one needed for this PR. The others should be removed

Comment (3xtr4t3rr3str14l) [reply packages/lodestar-cli/src/cmds/beacon/config.ts:35]: done.

Comment (3xtr4t3rr3str14l) [reply packages/lodestar-cli/src/cmds/beacon/config.ts:35]: done.

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**